app/models/registration_payment.rb

```diff
@@ -11,23 +11,43 @@ class RegistrationPayment < ApplicationRecord
 
   delegate :auto_accept_preference_live?, to: :registration
   after_create :auto_accept_hook, if: :auto_accept_preference_live?
-  after_create :auto_close_hook, unless: :refunded_registration_payment_id?
+  after_save :auto_close_hook
+
+  scope :completed, -> { where(is_completed: true) }
 
   monetize :amount_lowest_denomination,
            as: "amount",
            allow_nil: true,
            with_model_currency: :currency_code
 
+  def should_auto_close?
+    refunded_registration_payment_id.nil? && saved_change_to_is_completed?(to: true)
+  end
+
+  def create_uncaptured_payment
+    return unless self.is_completed?
+
+    RegistrationPayment.create(
+      amount_lowest_denomination: self.amount_lowest_denomination,
+      currency_code: self.currency_code,
+      user: self.user,
+      registration: self.registration,
+      is_completed: false,
+    )
+  end
+
   def amount_available_for_refund
-    amount_lowest_denomination + refunding_registration_payments.sum(:amount_lowest_denomination)
+    amount_lowest_denomination + refunding_registration_payments.completed.sum(:amount_lowest_denomination)
   end
 
   private def auto_accept_hook
     registration.attempt_auto_accept(:live)
   end
 
   private def auto_close_hook
-    registration.consider_auto_close
+    return unless refunded_registration_payment_id.nil? && is_completed?
+
+    registration.consider_auto_close if saved_change_to_is_completed? || previously_new_record?
```

Comment:
Why are all of these additional checks becoming relevant to `auto_close_hook`, but not to `auto_accept_hook`?

Reply:
You said:
> That said, I do think we should have a test that explicitly confirms that an is_completed: false registration_payment triggers a failed attempt_auto_accept - which I've added

which is interesting because my comment revolves around `attempt_auto_close`. Do we have sufficient similar test coverage there?

Reply:
We do, see:
- competition_spec:1917 (which auto_close relies on) 
- registration_spec:1243